genie-web/src/test/java/com/netflix/genie/web/controllers/RestControllerIntegrationTestsBase.java

```diff
@@ -382,4 +393,118 @@ void canDeleteTagForResource(
     private String getIdFromLocation(final String location) {
         return location.substring(location.lastIndexOf("/") + 1);
     }
+
+    private static String getLinkedResourceExpectedUri(
+        final String entityApi,
+        final String entityId,
+        final Set<String> optionalHalParams,
+        final String linkedEntityType) throws URISyntaxException {
+        final String uriPath = new StringBuilder()
+            .append(entityApi)
+            .append("/")
+            .append(entityId)
+            .append("/")
+            .append(linkedEntityType)
+            .toString();
+        final String uriString = new URI(URI_SCHEME, URI_HOST, uriPath, null).toString();
+
+        // Append HAL parameters separately to avoid URI encoding
+        String halParams = "";
```

Comment:
here it seems to me you can just make halParams a new `StringBuilder` and below you can do toString and it will have the same effect. Saves you from allocating effectively two strings